vignettes/programming_strategy.Rmd

```diff
@@ -739,6 +739,17 @@ Please take the following list as recommendation and try to adhere to its rules
 * `dplyr::if_else()` should be used when there are only two conditions. 
 Try to always set the `missing` argument whenever appropriate.
 
+## How Quoting is used
+* Some admiral arguments require selecting one particular option like `mode`, e.g. `mode = "last"`. Use quotation marks to capture these. The expected assertion function corresponding to these arguments is `assert_character_scalar()/assert_character_vector()`.
+* Many admiral arguments require capturing an expression which are to be evaluated _later_ inside the function body, see arguments like `new_vars`, e.g. `new_vars = exprs(TRTSDTM = EXSTDTM)`. Oftentimes, the assertion function corresponding to these are `assert_expr()/assert_expr_list()`
```

Comment:
I would distinguish two cases:
- Arguments like `new_var` or `filter` which expect a single variable or expression. Here the value is not quoted in the call. In the function it has to be quoted by using `enexpr()`. Usually this is combined with the assertion, e.g., `new_var <- assert_symbol(enexpr(new_var))`. These arguments are unquoted by using `!!`.
- Arguments like `new_vars` or `by_vars` which expect more than one variable or expression. Here the value is quoted in the call (by using `exprs()`). These arguments are unquoted by using `!!!`.

Reply:
added